core/src/tpu.rs

```diff
@@ -163,7 +164,7 @@ impl Tpu {
         _generator_config: Option<GeneratorConfig>, /* vestigial code for replay invalidator */
         key_notifiers: Arc<RwLock<KeyUpdaters>>,
         cancel: CancellationToken,
-    ) -> Self {
+    ) -> (Self, mpsc::Sender<BankingControlMsg>) {
```

Comment:
can we just pass in the receiver and pipe down instead of piping up? this pattern seems a bit odd

Reply:
98d61b362998d26525aad48efeb466a99aa738a7